Recipes/Recipes/All Recipes/Presenter/AllRecipesPresenter.swift

```diff
@@ -0,0 +1,176 @@
+//
+//  File.swift
+//  Recipes
+//
+//  Created by Vermut xxx on 04.03.2024.
+//
+
+import Foundation
+/// протокол вью  всех рецептов
+protocol RecipesViewProtocol: AnyObject {
+    func getRecipes(recipes: [Recipes])
+    func setTitle(_ title: String)
+    func goToTheCategory()
+    func reloadTableView()
+    func buttonTimeState(color: String, image: String)
+    func buttonCaloriesState(color: String, image: String)
+    func sortViewRecipes(recipes: [Recipes])
+
+}
+///протокол презентера
+protocol RecipeProtocol: AnyObject {
+    func getReceipts()
+    func goToRecipeDetails()
+    func goToCategory()
+    func getCategoryTitle()
+    func searchRecipes(text: String)
+    func checkSearch() -> [Recipes]
+    func startSearch()
+    func stopSearch()
+    func sortRecipes(category: [Recipes])
+
+
+}
+/// презентер всех рецептов
+final class AllRecipesPresenter {
+    private weak var view: RecipesViewProtocol?
+     weak var recipesCoordinator: RecipeCoordinator?
+    private var user: Recipes?
+    private var isSearching = false
+    private var searchNames: [Recipes] = []
+    private var recipes = Recipes.allRecipes
+    private var sortedCalories = SortedCalories.none
+    private var sortedTime = SortedTime.none
+    var sorted = Recipes.allRecipes
+
+
+    init(view: RecipesViewProtocol, coordinator: RecipeCoordinator) {
+        self.view = view
+        self.recipesCoordinator = coordinator
+    }
+    
+    func buttonCaloriesChange(category: [Recipes]) {
+            if sortedCalories == .none {
+                sortedCalories = .caloriesLow
+                view?.buttonCaloriesState(color: "background01", image: "filterLow")
```

Comment:
```suggestion
                view?.buttonCaloriesState(color: "background01", image: "filterLow")
```
в константы enum все строки всегда выносим "background01"